src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +164,213 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
                         idsToDelete.Add(queueItem.DocumentId);
                 }
             }
+        }
+        catch (ProduceException<string, byte[]> e)
+        {
+            if (e.Error.Code == ErrorCode.TopicAuthorizationFailed)
+            {
+                var topicErrorMessageBuilder = new StringBuilder(
+                    $"ETL process: {Name}. Failed to produce message.{Environment.NewLine}" +
+                    $"Add an required ACL permissions to your API Key that will make WRITE operations on the '{currentTopic.Name}' topic possible.");
+                
+                TryAbortTransactionAfterException(topicErrorMessageBuilder, out _);
+                throw new QueueLoadException(topicErrorMessageBuilder.ToString(), e);
+            }
 
-            _producer.CommitTransaction();
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Produce message operation failed. ");
```

Comment:
This is the initial error message. Can we move it to the top and adjust / append based on error code if needed. Something like:

```
var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Produce message operation failed. ");

if (e.Error.Code == ErrorCode.TopicAuthorizationFailed)
{
   errorMessageBuilder.Append($"Add an required ACL permissions to your API Key that will make WRITE operations on the '{currentTopic.Name}' topic possible.");
}
else if (TryToGetAclErrorsDetails(out var detailsMessage, currentTopic.Name))
{
    errorMessageBuilder.Append(detailsMessage);
}

if (TryAbortTransaction(out var abortException) == false)
{
   errorMessageBuilder.Append("Aborting Kafka transaction failed too.");
   e = new AggregateException(e, abortException);
}

throw new QueueLoadException(errorMessageBuilder.ToString(), e);

Reply:
done, thank you